include/platform/spi-slave.h

```diff
@@ -0,0 +1,171 @@
+/*
+ *  Copyright (c) 2016, Nest Labs, Inc.
+ *  All rights reserved.
+ *
+ *  Redistribution and use in source and binary forms, with or without
+ *  modification, are permitted provided that the following conditions are met:
+ *  1. Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *  2. Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *  3. Neither the name of the copyright holder nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/**
+ * @file spi-slave.h
+ * @brief
+ *   This file includes the platform abstraction for SPI slave communication.
+ */
+
+#ifndef SPI_SLAVE_H_
+#define SPI_SLAVE_H_
+
+#include <stdint.h>
+
+#include <openthread-types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup spi-slave SPI Slave
+ * @ingroup platform
+ *
+ * @brief
+ *   This module includes the platform abstraction for SPI slave communication.
+ *
+ * @{
+ *
+ */
+
+/**
+ * Indicates that a SPI transaction has completed with
+ * the given length. The data written to the slave has been
+ * written to the pointer indicated by the `anInputBuf` argument
+ * to the previous call to otPlatSpiSlavePrepareTransaction().
+ *
+ * Once this function is called, otPlatSpiSlavePrepareTransaction()
+ * is invalid and must be called again for the next transaction to be
+ * valid.
+ *
+ * Note that this function is always called at the end of a transaction,
+ * even if otPlatSpiSlavePrepareTransaction() has not yet been called.
+ *
+ * @param[in] aContext Context pointer passed into otPlatSpiSlaveEnable()
+ * @param[in] aLen     Length of the completed transaction, in bytes
+ */
+typedef void (*otPlatSpiSlaveTransactionCompleteCallback)(void *aContext, uint16_t aLen);
```

Comment:
The code which will be calling this API will be well aware of how SPI works and will ultimately not lose any data. It is important that this API reflect how SPI actually works and not try to do anything fancy like buffering.